operator/src/test/java/org/keycloak/operator/testsuite/integration/BaseOperatorTest.java

```diff
@@ -68,7 +68,7 @@ public abstract class BaseOperatorTest {
   public static final String TEST_RESULTS_DIR = "target/operator-test-results/";
   public static final String POD_LOGS_DIR = TEST_RESULTS_DIR + "pod-logs/";
 
-  public enum OperatorDeployment {local,remote}
+  public enum OperatorDeployment {local,remote,none}
```

Comment:
From my perspective, not deploying the Operator in an integration test is rather a special case than a regular `OperatorDeployment` type.

I would not probably go this way of introducing the `none` deployment type as it doesn't make sense for any of the integration tests but one. Thinking a bit more about this, isn't the `CRDTest` just a special case of a unit test? Downside is we'd need to instantiate the k8s client and apply CRDs and perform some cleanup after the test which we don't have a logic for there.

Another (a bit stupidly simple :)) alternative would be creating some `deployOperator` switch (either just a field in `BaseOperatorTest` or an annotation or something) to enable individual integration tests to skip the operator deployment/creation.

WDYT?

Reply:
> Thinking a bit more about this, isn't the CRDTest just a special case of a unit test?

It's still an integration test in that we need the full k8 support.  There may eventually be a drop in replacement for the fabric8 client crud mock based upon a partial kube or k3s replacement that could fill this niche more directly.

> Downside is we'd need to instantiate the k8s client and apply CRDs and perform some cleanup after the test which we don't have a logic for there.

This logic already exists in fabric8 via the KubernetesTest annotation.  However because of the Log usage in the BaseOperatorTest, calls like createCRDs can't be made unless it's also annotated as a QuarkusTest, which appearently doesn't play well with any other non-quarkus test extensions.  An alternative is to either duplicate crd creation method or remove the Log statements, let me see what you think of that.